### PR TITLE
yandex[major]: Add `@langchain/yandex` package

### PR DESCRIPTION
**Pull request by tyumentsev4**
This pull request adds yandex library with new Foundation Models API. Marks community modules as deprecated.

### CODE REVIEW

File: docs/core_docs/docs/integrations/chat/yandex.mdx [hunk 1]
```diff
@@ -4,6 +4,10 @@ sidebar_label: YandexGPT
 
 # ChatYandexGPT
 
+:::note
+This integration use deprecated [API v1alpha](https://cloud.yandex.ru/ru/docs/yandexgpt/api-ref/v1alpha/), please use the `@langchain/yandex` integration."
```
Comment (bracesproul) [inline docs/core_docs/docs/integrations/chat/yandex.mdx:8]:
```suggestion
This integration uses the deprecated [API v1alpha](https://cloud.yandex.ru/ru/docs/yandexgpt/api-ref/v1alpha/). Please use the `@langchain/yandex` integration."
```

File: docs/core_docs/docs/integrations/llms/yandex.mdx [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # YandexGPT
 
+:::note
+This integration use deprecated [API v1alpha](https://cloud.yandex.ru/ru/docs/yandexgpt/api-ref/v1alpha/), please use the `@langchain/yandex` integration."
```
Comment (bracesproul) [inline docs/core_docs/docs/integrations/llms/yandex.mdx:4]:
```suggestion
This integration uses the deprecated [API v1alpha](https://cloud.yandex.ru/ru/docs/yandexgpt/api-ref/v1alpha/). Please use the `@langchain/yandex` integration."
```

File: libs/langchain-community/src/chat_models/yandex.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ function _parseChatHistory(history: BaseMessage[]): [ParsedMessage[], string] {
 }
 
 /**
+ * @deprecated
```
Comment (bracesproul) [inline libs/langchain-community/src/chat_models/yandex.ts:41]:
```suggestion
 * @deprecated Prefer @langchain/yandex
```

File: libs/langchain-community/src/llms/yandex.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { LLM, type BaseLLMParams } from "@langchain/core/language_models/llms";
 
 const apiUrl = "https://llm.api.cloud.yandex.net/llm/v1alpha/instruct";
 
+/** @deprecated */
```
Comment (bracesproul) [inline libs/langchain-community/src/llms/yandex.ts:6]:
```suggestion
/** @deprecated Prefer @langchain/yandex */
```

File: libs/langchain-community/src/llms/yandex.ts [hunk 5]
```diff
@@ -32,6 +33,7 @@ export interface YandexGPTInputs extends BaseLLMParams {
   iamToken?: string;
 }
 
+/** @deprecated */
```
Comment (bracesproul) [inline libs/langchain-community/src/llms/yandex.ts:36]:
```suggestion
/** @deprecated Prefer @langchain/yandex */
```

File: libs/langchain-yandex/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+By default, the latest version of `yandexgpt-lite` is used from the folder specified in the parameter `folder_id` or `YC_FOLDER_ID` environment variable.
+
+### Examples
+
+```typescript
+import { ChatYandexGPT } from "@langchain/yandex";
+import { HumanMessage, SystemMessage } from "langchain/schema";
+
+const chat = new ChatYandexGPT();
+const response = await chat.call([
```
Comment (bracesproul) [inline libs/langchain-yandex/README.md:36]:
```suggestion
const response = await chat.invoke([
```

File: libs/langchain-yandex/README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+]);
+console.log(response)
+```
+
+```typescript
+import { YandexGPT } from "@langchain/yandex";
+
+const model = new YandexGPT();
+
+const res = await model.call('Translate "I love programming" into French.');
```
Comment (bracesproul) [inline libs/langchain-yandex/README.md:50]:
```suggestion
const res = await model.invoke(["Translate "I love programming" into French."]);
```

File: libs/langchain-yandex/package.json [hunk 8]
```diff
@@ -0,0 +1,101 @@
+{
+  "name": "yandex",
+  "version": "0.0.0",
+  "description": "Sample integration for LangChain.js",
```
Comment (bracesproul) [inline libs/langchain-yandex/package.json:4]:
```suggestion
  "name": "@langchain/yandex",
  "version": "0.0.0",
  "description": "Yandex integration for LangChain.js",
```

File: libs/langchain-yandex/src/tests/embeddings.int.test.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+test("Test GooglePalmEmbeddings.embedQuery", async () => {
+  const embeddings = new YandexGPTEmbeddings({
+    maxRetries: 1,
+  });
+  const res = await embeddings.embedQuery("Hello world");
+  expect(typeof res[0]).toBe("number");
+});
+
+test("Test GooglePalmEmbeddings.embedDocuments", async () => {
```
Comment (bracesproul) [inline libs/langchain-yandex/src/tests/embeddings.int.test.ts:12]:
```suggestion
test("Test YandexGPTEmbeddings.embedQuery", async () => {
  const embeddings = new YandexGPTEmbeddings({
    maxRetries: 1,
  });
  const res = await embeddings.embedQuery("Hello world");
  expect(typeof res[0]).toBe("number");
});

test("Test YandexGPTEmbeddings.embedDocuments", async () => {
```

**Review by bracesproul — COMMENTED** [review]
Thanks for doing this!! I committed a few changes, and will get to reviewing the rest in a little.

File: docs/core_docs/docs/integrations/chat/yandex.mdx [hunk 10]
```diff
@@ -19,6 +19,16 @@ Next, you have two authentication options:
 
 ## Usage
 
+You'll first need to install the `@langchain/yandex` package:
+
+:::tip
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/chat/yandex.mdx:24]: FYI just added a component for this:

https://github.com/langchain-ai/langchainjs/pull/3859/files#diff-4abc024fbd91edd4684e6c72069d70fa8dc66d051ee21ce675109344b10c218eR63

File: libs/langchain-yandex/src/embeddings.ts [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+      const embeddings: number[][] = [];
+
+      for (const text of texts) {
+        const bodyData = {
+          modelUri: this.modelURI,
+          text,
+        };
+
+        try {
+          const response = await fetch(apiUrl, {
```
Comment (jacoblee93) [inline libs/langchain-yandex/src/embeddings.ts:147]: nit: May want to consider concurrency here

File: libs/langchain-yandex/src/embeddings.ts [hunk 12]
```diff
@@ -0,0 +1,169 @@
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import { Embeddings, EmbeddingsParams } from "@langchain/core/embeddings";
+
+const apiUrl =
+  "https://llm.api.cloud.yandex.net/foundationModels/v1/textEmbedding";
+
+export interface YandexGPTEmbeddingsParams extends EmbeddingsParams {
```
Comment (jacoblee93) [inline libs/langchain-yandex/src/embeddings.ts:7]: I don't see any changed docs for embeddings?

**Comment by jacoblee93** [comment]
Looks good, a few nits but nothing blocking. Will merge tonight!

**Pull request merged**